### ISSUE TITLE
Slow content refresh

### ISSUE THREAD

**Issue by htmgarcia** [publishpress/PublishPress-Blocks#655]
The Content Display takes some time to display the actual content in Appearance > Widgets in WP 5.8 RC-4
This could be an issue introduced by us at some point to make sure when switching the filters for CPT are applied to the preview in regular post editor.
![Screen Shot 2021-07-16 at 11 19 24](https://user-images.githubusercontent.com/4999794/125978651-baf0825f-2fe8-44a0-a3a2-931a474933ff.png)
### Several seconds later displays the content:
![Screen Shot 2021-07-16 at 11 21 49](https://user-images.githubusercontent.com/4999794/125978729-b3f411f7-b24f-4bf3-87a5-dbb5a6dfa553.png)

**Comment by htmgarcia** [comment]
There is a delay of 5 or 6 seconds before the block start loading. This happens on Appearance > Widgets, but not in post edit.
`hasPosts` value is `0` until changes to false https://github.com/publishpress/PublishPress-Blocks/blob/ppb-655/src/assets/blocks/recent-posts/block.jsx#L836-L837
![Screen Shot 2021-07-29 at 15 23 29](https://user-images.githubusercontent.com/4999794/127560692-ab0f2263-685d-4e70-a9c6-46fd9c8aaa1d.png)

@contactashish13 any idea what could be causing this delay?

**Comment by contactashish13** [comment]
@htmgarcia on it today!

**Comment by contactashish13** [comment]
@htmgarcia the behavior is slightly different in posts and widgets. It appears that in widgets calling [getEntityRecords ](https://github.com/publishpress/PublishPress-Blocks/blob/ppb-655/src/assets/blocks/recent-posts/block.jsx#L1424) returns an empty array in widgets but a null in posts. That is why the transition from 0 to false/null to a positive number in widgets but in posts it goes from false/null to a positive number. 

Could it because because of cached data? I tried to add a random timestamp parameter `custom_pp` to `recentPostsQuery` to burst the cache but that doesn't work either. 

### in widgets
```
edit b4 getEntityRecords undefined {order: "desc", orderby: "date", per_page: 8, custom_pp: 1630400474133}
edit after getEntityRecords undefined {order: "desc", orderby: "date", per_page: 8, custom_pp: 1630400474133} []
```

### in posts
```
edit b4 getEntityRecords undefined {order: "desc", orderby: "date", per_page: 8, custom_pp: 1630400517659}
edit after getEntityRecords undefined {order: "desc", orderby: "date", per_page: 8, custom_pp: 1630400517659} null
```

with the below code change
```
console.log('edit b4 getEntityRecords', postType, recentPostsQuery);
let x = getEntityRecords( 'postType', postType ? postType : 'post', recentPostsQuery );
console.log('edit after getEntityRecords', postType, recentPostsQuery, x);
```

**Comment by htmgarcia** [comment]
@contactashish13 tweaking the code to detect when the block is in widget area or post edit. Would that help?

**Comment by contactashish13** [comment]
@htmgarcia yeah that was my first thought too but I couldn't find anything elegant that would allow that. I can always check if I'm inside the widget DOM but that seemed like a hack to me. Let me know if you have a better idea.

**Comment by htmgarcia** [comment]
Fixed with #800 and possibly in combination with #797
